File: solutions/julia/squeaky-clean/1/squeaky-clean.jl
function transform(ch)
    if ch=='-' 
        res = "_"
    elseif ch>='α' && ch<='ω' || ch>='ϐ' && ch<='ϛ'
        res = "?"
    elseif isspace(ch)
        res = ""
    elseif isletter(ch) && (ch == uppercase(ch))
        res = "-"*lowercase(ch)
    elseif isdigit(ch)
        res = ""
    else
        res = ""*ch
    end
end

function clean(str)
    res = ""
    for ch in str
        res = res*transform(ch)
    end
    res
end

File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String in) {
        char[] input = in.toCharArray();
        StringBuilder out = new StringBuilder();
        String leet = "43017";
        for(int i=0; i<input.length; i++){
            char ch = input[i];
            if(ch==' '){
                out.append('_');
                continue;
            } 
            else if (ch=='-'){
                if (i+1<input.length){
                    i++;
                    char c = input[i];
                    if(Character.isLetter(c)){
                        out.append(                            Character.toUpperCase(c));}
                }                
            }
            else if(leet.indexOf(ch)!=-1){
                int leet_idx = leet.indexOf(ch);
                out.append("aeolt".charAt(leet_idx));
            }
            else if(Character.isLetter(ch) || Character.isDigit(ch)){
                out.append(ch);    
            }
           /* else{
                continue;    
            }*/        
        }
        return out.toString();
    }
    
}
